dotnet-interactive/Program.cs

```diff
@@ -113,6 +115,27 @@ int GetFreePort()
                 l.Stop();
                 return port;
             }
+
+            int GetFreePortInRange(PortRange httpPortRange)
+            {
+                for (var currentPort = httpPortRange.Start; currentPort <= httpPortRange.End; currentPort++)
+                {
+                    try
+                    {
+                        var l = new TcpListener(IPAddress.Loopback, currentPort);
+                        l.Start();
+                        var port = ((IPEndPoint)l.LocalEndpoint).Port;
+                        l.Stop();
+                        return currentPort;
+                    }
```

Comment:
Rather than duplicate this code, can we consolidate these into a single method with an optional parameter for the port range?